src/cli/upgrade_command.zig

```diff
@@ -570,9 +570,17 @@ pub const UpgradeCommand = struct {
 
                     var buf: bun.PathBuffer = undefined;
                     const powershell_path =
-                        bun.which(&buf, bun.getenvZ("PATH") orelse "", "", "powershell") orelse
+                        bun.which(&buf, bun.env_var.path.get() orelse "", "", "powershell") orelse
                         hardcoded_system_powershell: {
-                            const system_root = bun.getenvZ("SystemRoot") orelse "C:\\Windows";
+                            var windir = switch (bun.os.win32.WinDir.query(ctx.allocator)) {
+                                .err => {
+                                    Output.prettyErrorln("<r><red>error:<r> Failed to unzip {s} due to PowerShell not being installed.", .{tmpname});
+                                    Global.exit(1);
+                                },
+                                .result => |v| v,
+                            };
+                            defer windir.deinit();
+                            const system_root = windir.slice();
```

Comment:
system_root used to be hardcoded to `getenv(SystemRoot) orelse Windows`.  should it default to that still in the error case?

It looks like there's a test covering 'bun upgrade' in CI so if that passes hopefully it means it's fine